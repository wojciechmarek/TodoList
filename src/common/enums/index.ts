export * from './theme';
export * from './task-status';
