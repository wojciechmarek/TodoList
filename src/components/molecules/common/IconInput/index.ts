export * from './icon-input';
export * from './icon-input.interface';
