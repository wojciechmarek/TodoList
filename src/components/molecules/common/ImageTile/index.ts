export * from './image-tile';
export * from './image-tile.interface';
