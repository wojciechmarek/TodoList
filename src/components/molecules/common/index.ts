export * from './IconButton';
export * from './IconInput';
export * from './ImageTile';
