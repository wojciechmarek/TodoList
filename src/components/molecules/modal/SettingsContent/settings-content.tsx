import { useState } from 'react';
import { ImageTile } from '../../common';
import { ModalSettingsContentProps } from './settings-content.interface';
import { SettingsContainer } from './settings-content.styled';
import { Theme } from '../../../../common';

export const ModalSettingsContent = ({
  theme,
  handleThemeClick,
}: ModalSettingsContentProps) => {
  const [currentTheme, setCurrentTheme] = useState(theme);

  const handleClick = (theme: Theme) => {
    handleThemeClick(theme);
    setCurrentTheme(theme);
  };

  return (
    <SettingsContainer>
      <ImageTile
        isActive={currentTheme === Theme.ModernDark}
        text="Modern flat dark"
        handleThemeClick={() => handleClick(Theme.ModernDark)}
      />
      <ImageTile
        isActive={currentTheme === Theme.ModernLight}
        text="Modern flat light"
        handleThemeClick={() => handleClick(Theme.ModernLight)}
      />
      <ImageTile
        isActive={currentTheme === Theme.Bruthalism}
        text="Bruthalism"
        handleThemeClick={() => handleClick(Theme.Bruthalism)}
      />
      <ImageTile
        isActive={currentTheme === Theme.Neobruthalism}
        text="Neobruthalism"
        handleThemeClick={() => handleClick(Theme.Neobruthalism)}
      />
      <ImageTile
        isActive={currentTheme === Theme.Cardboard}
        text="Cardboard"
        handleThemeClick={() => handleClick(Theme.Cardboard)}
      />
      <ImageTile
        isActive={currentTheme === Theme.Glassmorphism}
        text="Glassmorphism"
        handleThemeClick={() => handleClick(Theme.Glassmorphism)}
      />
      <ImageTile
        isActive={currentTheme === Theme.Claymorphism}
        text="Claymorphism"
        handleThemeClick={() => handleClick(Theme.Claymorphism)}
      />
      <ImageTile
        isActive={currentTheme === Theme.Neumorphism}
        text="Neumorphism"
        handleThemeClick={() => handleClick(Theme.Neumorphism)}
      />
      <ImageTile
        isActive={currentTheme === Theme.Aurora}
        text="Aurora"
        handleThemeClick={() => handleClick(Theme.Aurora)}
      />
      <ImageTile
        isActive={currentTheme === Theme.AppleCupertino}
        text="Apple Cupertino"
        handleThemeClick={() => handleClick(Theme.AppleCupertino)}
      />
      <ImageTile
        isActive={currentTheme === Theme.GoogleMaterial}
        text="Google Material"
        handleThemeClick={() => handleClick(Theme.GoogleMaterial)}
      />
    </SettingsContainer>
  );
};
