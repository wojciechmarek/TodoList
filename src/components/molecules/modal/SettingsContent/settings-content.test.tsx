import { act, configure, render, renderHook } from '@testing-library/react';
import { ModalSettingsContentProps } from './settings-content.interface';
import { UseFormReturn, useForm } from 'react-hook-form';
import { Task, TaskStatus, Theme } from '../../../../common';
import { ModalSettingsContent } from './settings-content';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';

describe('SettingsContent', () => {
  let modalSettingsContentProps: ModalSettingsContentProps;

  beforeEach(() => {
    configure({ throwSuggestions: true });

    modalSettingsContentProps = {
      theme: Theme.Cardboard,
      handleThemeClick: jest.fn(),
    };
  });

  it('should render', () => {
    const { baseElement } = render(
      <ModalSettingsContent {...modalSettingsContentProps} />
    );
    expect(baseElement).toBeTruthy();
  });

  describe('when the edit settings modal content is rendered', () => {
    it('should render 11 theme tiles', () => {
      const { getAllByTestId } = render(
        <ModalSettingsContent {...modalSettingsContentProps} />
      );
      const themeTiles = getAllByTestId('tile');
      expect(themeTiles.length).toBe(11);
    });
  });

  describe('when the tile is clicked', () => {
    it('should call the handleThemeClick function', () => {
      const { getAllByTestId } = render(
        <ModalSettingsContent {...modalSettingsContentProps} />
      );
      const themeTiles = getAllByTestId('tile');
      const tile = themeTiles.at(0);
      act(() => {
        tile?.click();
      });
      expect(modalSettingsContentProps.handleThemeClick).toHaveBeenCalled();
    });

    it('should call the handleThemeClick function with correct parameter', () => {
      const { getAllByTestId } = render(
        <ModalSettingsContent {...modalSettingsContentProps} />
      );
      const themeTiles = getAllByTestId('tile');

      // Neobruthalism theme at index 3
      const tile = themeTiles.at(3);

      act(() => {
        tile?.click();
      });
      expect(modalSettingsContentProps.handleThemeClick).toHaveBeenCalledWith(
        Theme.Neobruthalism
      );
    });
  });
});
