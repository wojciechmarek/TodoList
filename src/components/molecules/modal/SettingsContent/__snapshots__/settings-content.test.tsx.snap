// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsContent should match the snapshot 1`] = `
.emotion-0 {
  width: calc(100% - 4em);
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: grid;
  -webkit-column-gap: 0.1em;
  column-gap: 0.1em;
  margin: 2em;
  grid-template-columns: auto auto;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: flex-start;
  align-content: flex-start;
  overflow: auto;
}

.emotion-1 {
  margin: 0.75em;
  height: 5em;
  width: 10em;
  display: grid;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: var(--theme-tile-button-bg-color);
  border-radius: 0.5em;
  color: var(--button-font-color);
  font-weight: 800;
  border: var(--standard-border);
  border-radius: var(--standard-border-radius);
  box-shadow: var(--container-box-shadow);
  cursor: pointer;
  -webkit-transition: filter 0.3s ease-in-out;
  transition: filter 0.3s ease-in-out;
  -webkit-filter: none;
  filter: none;
}

.emotion-1:hover {
  -webkit-filter: brightness(0.75);
  filter: brightness(0.75);
}

.emotion-5 {
  margin: 0.75em;
  height: 5em;
  width: 10em;
  display: grid;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: var(--theme-tile-button-bg-color);
  border-radius: 0.5em;
  color: var(--button-font-color);
  font-weight: 800;
  border: var(--standard-border);
  border-radius: var(--standard-border-radius);
  box-shadow: var(--container-box-shadow);
  cursor: pointer;
  -webkit-transition: filter 0.3s ease-in-out;
  transition: filter 0.3s ease-in-out;
  -webkit-filter: brightness(0.8);
  filter: brightness(0.8);
}

.emotion-5:hover {
  -webkit-filter: brightness(0.75);
  filter: brightness(0.75);
}

<div
  className="emotion-0"
>
  <div
    className="emotion-1"
    data-testid="tile"
    onClick={[Function]}
  >
    Modern flat dark
  </div>
  <div
    className="emotion-1"
    data-testid="tile"
    onClick={[Function]}
  >
    Modern flat light
  </div>
  <div
    className="emotion-1"
    data-testid="tile"
    onClick={[Function]}
  >
    Bruthalism
  </div>
  <div
    className="emotion-1"
    data-testid="tile"
    onClick={[Function]}
  >
    Neobruthalism
  </div>
  <div
    className="emotion-5"
    data-testid="tile"
    onClick={[Function]}
  >
    Cardboard
  </div>
  <div
    className="emotion-1"
    data-testid="tile"
    onClick={[Function]}
  >
    Glassmorphism
  </div>
  <div
    className="emotion-1"
    data-testid="tile"
    onClick={[Function]}
  >
    Claymorphism
  </div>
  <div
    className="emotion-1"
    data-testid="tile"
    onClick={[Function]}
  >
    Neumorphism
  </div>
  <div
    className="emotion-1"
    data-testid="tile"
    onClick={[Function]}
  >
    Aurora
  </div>
  <div
    className="emotion-1"
    data-testid="tile"
    onClick={[Function]}
  >
    VFD Display
  </div>
  <div
    className="emotion-1"
    data-testid="tile"
    onClick={[Function]}
  >
    CRT
  </div>
</div>
`;
